rules/DowngradePhp70/Rector/Instanceof_/DowngradeInstanceofThrowableRector.php

```diff
@@ -0,0 +1,132 @@
+                    <<<'CODE_SAMPLE'
+return ($throwable = $e) instanceof \Throwable || $throwable instanceof \Exception;
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Instanceof_::class];
+    }
+
+    /**
+     * @param Instanceof_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! ($node->class instanceof Name && $this->nodeNameResolver->isName($node->class, 'Throwable'))) {
+            return null;
+        }
+        $createdByRule = $node->getAttribute(AttributeKey::CREATED_BY_RULE) ?? [];
+        if (in_array(self::class, $createdByRule, true)) {
+            return null;
+        }
+
+        // Ensure the refactoring is idempotent.
+        if ($this->isAlreadyTransformed($node)) {
+            return null;
+        }
+
+        $variable = $this->createVariable($node);
+        $throwableCheck = new Instanceof_(new Assign($variable, $node->expr), $node->class);
+        $exceptionFallbackCheck = $this->createFallbackCheck($variable);
+        $expression = new BooleanOr($throwableCheck, $exceptionFallbackCheck);
+
+        return $expression;
+    }
+
+    private function createVariable(Instanceof_ $expr): Variable
+    {
+        $currentStmt = $expr->getAttribute(AttributeKey::CURRENT_STATEMENT);
+        $scope = $currentStmt->getAttribute(AttributeKey::SCOPE);
+
+        return new Variable($this->variableNaming->createCountedValueName('throwable', $scope));
+    }
+
+    /**
+     * Also checks similar manual transformations.
+     */
+    private function isAlreadyTransformed(Instanceof_ $instanceof_): bool
+    {
+        /** @var Node $parentNode */
+        $parentNode = $instanceof_->getAttribute(AttributeKey::PARENT_NODE);
+        if (! $parentNode instanceof BooleanOr) {
+            return false;
+        }
+
+        $hasVariableToFindInDisjunction = ($var = $instanceof_->expr) instanceof Variable || ($instanceof_->expr instanceof Assign && ($var = $instanceof_->expr->var) instanceof Variable);
+        if (! $hasVariableToFindInDisjunction) {
+            return false;
+        }
+
+        $disjunctionTree = $this->binaryOpTreeRootLocator->findOperationRoot($instanceof_, BooleanOr::class);
+        $disjuncts = $this->binaryOpConditionsCollector->findConditions($disjunctionTree, BooleanOr::class);
+
+        // If we transformed it ourselves, the second check can only be to the right
+        // since it uses the assigned variable.
+        if ($instanceof_->expr instanceof Assign) {
+            $index = array_search($instanceof_, $disjuncts, true);
+            if ($index !== false) {
+                $disjuncts = array_slice($disjuncts, $index);
+            }
+        }
+
+        $expectedDisjunct = $this->createFallbackCheck($var);
+        return $this->nodeComparator->isNodeEqual($expectedDisjunct, $disjuncts);
```

Comment:
With this extended algorithm, it will also skip `$e instanceof \Exception || ($e = $e->getPrevious()) || $e instanceof Throwable`, which the previous variant would correctly change to `$e instanceof \Exception || ($e = $e->getPrevious()) || ($e instanceof Throwable || $e instanceof \Exception)`